src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -159,179 +109,283 @@ public IDisposable On(string methodName, Type[] parameterTypes, Func<object[], o
             return new Subscription(invocationHandler, invocationList);
         }
 
-        public async Task<ChannelReader<object>> StreamAsChannelAsync(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default)
+        public async Task<ChannelReader<object>> StreamAsChannelAsync(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default) =>
+            await StreamAsChannelAsyncCore(methodName, returnType, args, cancellationToken).ForceAsync();
+
+        public async Task<object> InvokeAsync(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default) =>
+            await InvokeAsyncCore(methodName, returnType, args, cancellationToken).ForceAsync();
+
+        // REVIEW: We don't generally use cancellation tokens when writing to a pipe because the asynchrony is only the result of backpressure.
+        // However, this would be the only "invocation" method _without_ a cancellation token... which is odd.
```

Comment:
We can still pass it to flush to cancel the flush if it *is* async

Reply:
I'm going to need to clarify your position on `CancellationToken` and Pipelines; I'm getting conflicting messages ;)

Reply:
I don't want to flow it everywhere but cases like this we can pass it directly to the pipe API and let it do what it does.